app/services/ccms/add_case_service.rb

```diff
@@ -0,0 +1,26 @@
+module CCMS
+  class AddCaseService < BaseSubmissionService
+    def call
+      response = case_add_requestor.call
+      if case_add_response_parser(response).success?
+        submission.case_add_transaction_id = case_add_requestor.transaction_request_id
+        create_history(submission.aasm_state, :case_submitted)
```

Comment:
Would this work? :- 
```ruby
create_history(submission.aasm_state, :case_submitted) if submission.submit_case!
```
That way the history will only be created if the case submission is successful.

Reply:
Good point. That will work, with a minor change to the values passed to `create_history`, as the status is now being updated before the history record is created, rather than afterwards.

That pattern is used in several other classes, so I'll push a separate commit changing them too.